src/main/osd/osd_elements.c

```diff
@@ -2107,6 +2109,66 @@ bool osdDrawNextActiveElement(displayPort_t *osdDisplayPort, timeUs_t currentTim
     return retval;
 }
 
+#ifdef USE_SPEC_PREARM_SCREEN
+void osdDrawSpec(displayPort_t *osdDisplayPort)
+{
+    if (!ARMING_FLAG(ARMED) && osdConfig()->osd_show_spec_prearm)
+    {
+        const uint8_t midRow = osdDisplayPort->rows / 2;
+        const uint8_t midCol = osdDisplayPort->cols / 2;
+
+        char buff[OSD_ELEMENT_BUFFER_LENGTH] = "";
+
+        memset(buff,0,strlen(buff));
+        int len = 0;
+        int currentRow = midRow - 3;
+
+        bool rpmLimitActive = false;
+
+#ifdef USE_RPM_LIMIT
+        rpmLimitActive = mixerConfig()->rpm_limit > 0 && isProtocolBidirectionalDshot(&motorConfig()->dev);
+        if (rpmLimitActive) {
+            len = tfp_sprintf(buff, "RPM LIMIT ON  %d", mixerConfig()->rpm_limit_value);
+        } else {
+            len = tfp_sprintf(buff, "%s", "RPM LIMIT OFF");
+        }
+        displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
+
+        if (rpmLimitActive) {
+            memset(buff,0,strlen(buff));
+            len = tfp_sprintf(buff, "KV %d   POLES %d", motorConfig()->kv, motorConfig()->motorPoleCount);
+            displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
+
+            memset(buff,0,strlen(buff));
+            len = tfp_sprintf(buff, "%d  %d  %d", mixerConfig()->rpm_limit_p, mixerConfig()->rpm_limit_i, mixerConfig()->rpm_limit_d);
+            displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
+        }
+#endif // #USE_RPM_LIMIT
+
+        if (!rpmLimitActive) {
```

Comment:
```suggestion
        } else
#endif // #USE_RPM_LIMIT
        {
```

Reply:
i think this won't work.
The section that prints THR LIMIT should be shown when rpm limit is enabled or when there is no rpm limit in the code.

Reply:
This should work. Try it out. `rpmLimitActive` still gets checked (see my otherr sugestion below)